src/ToricVarieties/AttributesAndMethods.jl

```diff
@@ -1,50 +1,345 @@
+
+Compute the i-th Betti number of the normal toric variety `v`.
+"""
+function ith_betti_number(v::AbstractNormalToricVariety, i::Int)
+    if isodd(i)
+        return 0
+    end
+    k = div(i, 2)
+    f_vector = Vector{Int}(pm_ntv(v).F_VECTOR)
+    pushfirst!(f_vector, 1)
+    betti_number = sum((-1)^(i-k) * binomial(i,k) * f_vector[dim(v) - i + 1] for i=k:dim(v))
+    return betti_number
+    
+end
+export ith_betti_number
+
+
+@doc Markdown.doc"""
+    euler_characteristic(v::AbstractNormalToricVariety)
 
 Computes the Euler characteristic of the normal toric variety `v`.
 """
-function euler_characteristic( v::AbstractNormalToricVariety )
-    f_vector = Vector{Int}(v.polymakeNTV.F_VECTOR)
-    return f_vector[ dim( v ) ]
+function euler_characteristic(v::AbstractNormalToricVariety)
+    f_vector = Vector{Int}(pm_ntv(v).F_VECTOR)
+    return f_vector[dim(v)]
 end
 export euler_characteristic
 
 
+############################
+# Rings and ideals
+############################
+
+
+@doc Markdown.doc"""
+    cox_ring(v::AbstractNormalToricVariety)
+
+Computes the Cox ring of the normal toric variety `v`.
+
+# Examples
+```jdoctest
+julia> p2 = toric_projective_space(2)
+A normal toric variety corresponding to a polyhedral fan in ambient dimension 2
+
+julia> cox_ring(p2)
+(Multivariate Polynomial Ring in x[1], x[2], x[3] over Rational Field graded by 
+  x[1] -> [0 0 1]
+  x[2] -> [0 0 1]
+  x[3] -> [0 0 1], MPolyElem_dec{fmpq, fmpq_mpoly}[x[1], x[2], x[3]])
+```
+"""
+function cox_ring(v::AbstractNormalToricVariety)
+    Qx, x = PolynomialRing(QQ, :x=>1:pm_ntv(v).N_RAYS)
+    return grade(Qx,gens(class_group(v)))[1]
+end
+export cox_ring
+
+
+@doc Markdown.doc"""
+    list_of_variables_of_cox_ring(v::NormalToricVariety)
+
+Lists the names of the indeterminates of the Cox ring of a normal toric variety `v`.
+
+# Examples
+```jdoctest
+julia> p2 = toric_projective_space(2)
+A normal toric variety corresponding to a polyhedral fan in ambient dimension 2
+
+julia> list_of_variables_of_cox_ring(p2)
+3-element Vector{MPolyElem_dec{fmpq, fmpq_mpoly}}:
+ x1
+ x2
+ x3
+```
+"""
+function list_of_variables_of_cox_ring(v::AbstractNormalToricVariety)
+    return gens(cox_ring(v))
+end
+export list_of_variables_of_cox_ring
+
+
+@doc Markdown.doc"""
+    stanley_reisner_ideal(v::NormalToricVariety)
+Computes the Stanley-Reisner ideal of a normal toric variety `v`.
+
+# Examples
+```jdoctest
+julia> p2 = toric_projective_space(2)
+A normal toric variety corresponding to a polyhedral fan in ambient dimension 2
+
+julia> length(stanley_reisner_ideal(P2).gens)
+1
+```
+"""
+function stanley_reisner_ideal(v::AbstractNormalToricVariety)
+    collections = primitive_collections(fan_of_variety(v))
+    SR_generators = []
+    vars = list_of_variables_of_cox_ring(v)
+    for I in collections
+        buffer = vars[I[1]]
+        for k in 2 : length(I)
+           buffer = buffer * vars[I[k]]
+        end
+        push!(SR_generators, buffer)
+    end
```

Comment:
Could perhaps be rewritten as
```suggestion
    for I in collections
        push!(SR_generators, prod(vars[I]))
    end
```
or even
```suggestion
    SR_generators = [prod(vars[I]) for I in collections]
```

Reply:
Yes, that is a very good simplification. Thank you @fingolfin .